File: src/main/java/com/thesilentnights/easylogin/repo/PlayerCache.java
package com.thesilentnights.easylogin.repo;

import com.thesilentnights.easylogin.pojo.PlayerAccount;

import java.util.Map;
import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

public class PlayerCache {

    private static final Map<UUID, PlayerAccount> cacheMap = new ConcurrentHashMap<>();

    public static void addAccount(PlayerAccount account) {
        cacheMap.put(account.getUuid(), account);
    }

    public static Optional<PlayerAccount> getAccount(UUID uuid) {
        return Optional.ofNullable(cacheMap.get(uuid));
    }

    public static boolean hasAccount(UUID uuid) {
        return cacheMap.containsKey(uuid);
    }

    public static void dropAccount(UUID uuid, boolean tempDrop) {
        if (tempDrop) {
            PlayerAccount uuid1 = cacheMap.get(uuid);
            if (uuid1 != null) {
                PlayerSessionCache.scheduleDrop(uuid1);
            }
        }
        cacheMap.remove(uuid);
    }
}

File: src/main/kotlin/com/thesilentnights/easylogin/repo/PlayerCache.kt
package com.thesilentnights.easylogin.repo

import com.thesilentnights.easylogin.pojo.PlayerAccount
import java.util.*
import java.util.concurrent.ConcurrentHashMap

object PlayerCache {
    private val cacheMap: MutableMap<UUID, PlayerAccount> = ConcurrentHashMap<UUID, PlayerAccount>()

    fun addAccount(account: PlayerAccount) {
        cacheMap[account.uuid] = account
    }

    fun getAccount(uuid: UUID): Optional<PlayerAccount> {
        return Optional.ofNullable(cacheMap[uuid])
    }

    fun hasAccount(uuid: UUID): Boolean {
        return cacheMap.containsKey(uuid)
    }

    fun dropAccount(uuid: UUID, tempDrop: Boolean) {
        if (tempDrop) {
            val uuid1: PlayerAccount = cacheMap.get(uuid)!!
            PlayerSessionCache.scheduleDrop(uuid1)
        }
        cacheMap.remove(uuid)
    }
}
